Add support TypeScript

0xlay opened the pull request:
Support TypeScript was added: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10353#issue-963344194

donho commented:
the extension "ts" is recognized as JavaScript, which applies to Cpp lexer also. What makes your PR better?
A screenshot of Before and After could help.

0xlay commented:
@doho TypeScript has extensions ts, tsx. JavaScript has extensions js, jsx. These are two different languages. TypeScript has static typing, but JavaScript does not.
![](https://imgur.com/Zq040F3.png)
My PR adds TypeScript lang as a new language.

![](https://imgur.com/3q06JVk.png)

donho commented:
> My PR adds TypeScript lang as a new language.

Yes, I do understand what you have added in your PR. However, besides adding  TypeScript lang, I don't see the advantage of syntax highlighting in TypeScript.

0xlay commented:
@donho I added new keywords to the "langs.model.xml" file for the TypeScript lang that are not in JavaScript. And some of the keywords I removed. I think the logical solution would be to add TypeScript as a new language, rather than mixing everything with one JavaScript. But it's decide you.

donho commented on PowerEditor/src/Notepad_plus.cpp:
```diff
@@ -3448,6 +3448,8 @@ LangType Notepad_plus::menuID2LangType(int cmdID)
         case IDM_LANG_VISUALPROLOG:
             return L_VISUALPROLOG;
+		case IDM_LANG_TYPESCRIPT:
+			return L_TYPESCRIPT;
```
Indentation should be aligned with other lines.

donho commented on PowerEditor/visual.net/notepadPlus.vcxproj:
```diff
@@ -530,6 +530,11 @@ copy ..\src\contextMenu.xml ..\binarm64\contextMenu.xml
+    <Image Include="..\src\icons\asciiPanel2.ico" />
+    <Image Include="..\src\icons\chameleon.ico" />
+    <Image Include="..\src\icons\clipboardPanel.ico" />
+    <Image Include="..\src\icons\clipboardPanel2.ico" />
```
All the diff in `notepadPlus.vcxproj` should be restored.
There's no reason to modify project file.

donho reviewed (changes requested):
Please correct the 2 requested points.

0xlay commented:
@donho This project has a code style problem (mixing tabs and spaces). GitHub doesn't like the mixing of tabs and spaces.
![](https://imgur.com/pihHjcK.png)

donho commented:
@0xlay 
I just realized that in dark mode there's no syntax highlighting in TypeScript document, do you mind to add it in `DarkModeDefault.xml` please ?

donho commented:
@0xlay 
Another request for auto-completion & function list of type script - but it could be done later.

0xlay commented:
@donho Ok. I will add in next PR support dark theme and auto-completion & function list of type script
